src/plugin_proc.h

```diff
@@ -1,31 +1,235 @@
+ */
 extern int do_proc_net_rpc_nfsd(int update_every, usec_t dt);
+/**
+ * Data collector of `/proc/sys/kernel/random/entropy_avail`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_sys_kernel_random_entropy_avail(int update_every, usec_t dt);
+/**
+ * Data collector of `/proc/interrupts`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_interrupts(int update_every, usec_t dt);
+/**
+ * Data collector of `/proc/softirqs`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_softirqs(int update_every, usec_t dt);
+/**
+ * Data collector of `/sys/kernel/mm/ksm`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_sys_kernel_mm_ksm(int update_every, usec_t dt);
+/**
+ * Data collector of `/proc/loadavg`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_loadavg(int update_every, usec_t dt);
+/**
+ * Data collector of `/proc/net/stat/synproxy`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_net_stat_synproxy(int update_every, usec_t dt);
+/**
+ * Data collector of `/proc/net/dev`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_net_softnet_stat(int update_every, usec_t dt);
+/**
+ * Data collector of `/proc/net/softnet_stat`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_uptime(int update_every, usec_t dt);
+/**
+ * Data collector of `/sys/devices/system/edac/mc`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_sys_devices_system_edac_mc(int update_every, usec_t dt);
+/**
+ * Data collector of `/sys/devices/system/node`.
+ *
+ * @param update_every intervall in seconds this is called.
+ * @param dt microseconds passed since the last call.
+ * @return 0 on success. 1 on error.
+ */
 extern int do_proc_sys_devices_system_node(int update_every, usec_t dt);
 
+/**
+ * ktsaou: Your help needed
+ *
+ * @return numa_node_count
+ */
 extern int get_numa_node_count(void);
```

Comment:
@tycho 

I think this is used to find the number of numa nodes in this system.